IPython/html/static/notebook/js/notebook.js

```diff
@@ -663,10 +663,9 @@ define(function (require) {
     Notebook.prototype.command_mode = function () {
         var cell = this.get_cell(this.get_edit_index());
         if (cell && this.mode !== 'command') {
-            // We don't call cell.command_mode, but rather call cell.focus_cell()
-            // which will blur and CM editor and trigger the call to
-            // handle_command_mode.
-            cell.focus_cell();
+            // We don't call cell.command_mode, but rather blur the CM editor
+            // which will trigger the call to handle_command_mode.
+            cell.code_mirror.getInputField().blur();
```

Comment:
I am also pretty wary about this change from calling focus_cell to directly bluring the code mirror input field. I seem to remember that there were very subtle bugs that we found when we called blur directly. I probably prefer to revert it this close to 3.1. Longer term I think it would be a good idea to play around with this approach.